src/main.tf

```diff
@@ -35,7 +36,7 @@ resource "azurerm_healthcare_fhir_service" "main" {
     # authority is used for token validation
     authority = "https://login.microsoftonline.com/${var.azure_service_principal.data.tenant_id}"
     # audience identifies intended recipients of the token
-    audience = "https://${var.md_metadata.name_prefix}.fhir.azurehealthcareapis.com"
+    audience = "https://${azurerm_healthcare_workspace.main.name}-${var.md_metadata.name_prefix}.fhir.azurehealthcareapis.com"
```

Comment:
Adding a prefix here reminds me of the DNS for app-service and function-app. 👍 

This is interesting though. The authority is active directory for this tenant, and the audience is important. I wonder if this has any influence on RBAC for this service.

Reply:
I'll find out and let you know!